pkg/cloudprovider/providers/oci/instances.go

```diff
@@ -92,6 +92,7 @@ func (cp *CloudProvider) NodeAddresses(ctx context.Context, name types.NodeName)
 // nodeaddresses are being queried. i.e. local metadata services cannot be used
 // in this method to obtain nodeaddresses.
 func (cp *CloudProvider) NodeAddressesByProviderID(ctx context.Context, providerID string) ([]api.NodeAddress, error) {
+	addresses := []api.NodeAddress{}
```

Comment:
You don't need this variable. You can use vnicAddresses below to build your list of addresses.

Reply:
Done in the new commit